### PR TITLE
fix #9609 feat(cirrus): Api to support health checks

### LINKED ISSUE #9609: Container health-checks Dockerflow for Cirrus
┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3934)

### PR DESCRIPTION
**Pull request by yashikakhurana**
Because

- We want to configure Liveness and Readiness probes for the Cirrus container

This commit

- Add two API endpoints and return status 200 
```
/__lbheartbeat__
/__heartbeat__

```

fixes #9609

### CODE REVIEW

**Comment by yashikakhurana** [comment]
Due to some changes in the users and groups in docker file, I am getting this error when I do `make cirrus_up` `
experimenter-cirrus-1  | ERROR:glean:Failed to initialize Glean: An I/O error occurred: Permission denied (os error 13)`

**Review by jaredlockhart — COMMENTED** [review]
I'm unable to test this locally because of https://github.com/mozilla/experimenter/issues/9625

**Comment by jaredlockhart** [comment]
Settings for testing

```
CIRRUS_REMOTE_SETTING_URL=http://kinto:8888/v1/buckets/main/collections/nimbus-web-experiments/records
CIRRUS_REMOTE_SETTING_REFRESH_RATE_IN_SECONDS=10
CIRRUS_APP_ID=test_app_id
CIRRUS_APP_NAME=test_app_name
CIRRUS_CHANNEL=developer
CIRRUS_FML_PATH=./tests/feature_manifest/sample.fml.yaml
CIRRUS_SENTRY_DSN=
CIRRUS_REMOTE_SETTING_REFRESH_RATE_IN_SECONDS=10
```

**Comment by jaredlockhart** [comment]
I get this

```py
INFO:     Uvicorn running on http://0.0.0.0:8001 (Press CTRL+C to quit)
ERROR:cirrus.experiment_recipes:Failed to fetch recipes: 401 Client Error: Unauthorized for url: http://kinto:8888/v1/buckets/main/collections/nimbus-web-experiments/records
ERROR:cirrus.main:Failed to fetch recipes: 401 Client Error: Unauthorized for url: http://kinto:8888/v1/buckets/main/collections/nimbus-web-experiments/records
ERROR:apscheduler.executors.default:Job "fetch_schedule_recipes (trigger: interval[0:00:10], next run at: 2023-10-23 20:24:57 UTC)" raised an exception
Traceback (most recent call last):
  File "/app/cirrus/main.py", line 215, in fetch_schedule_recipes
    app.state.remote_setting.fetch_recipes()
  File "/app/cirrus/experiment_recipes.py", line 49, in fetch_recipes
    raise e
  File "/app/cirrus/experiment_recipes.py", line 46, in fetch_recipes
    response.raise_for_status()  # raises an HTTPError for bad status codes
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: http://kinto:8888/v1/buckets/main/collections/nimbus-web-experiments/records

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/apscheduler/executors/base_py3.py", line 30, in run_coroutine_job
    retval = await job.func(*job.args, **job.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/cirrus/main.py", line 219, in fetch_schedule_recipes
    app.state.scheduler.add_job(
  File "/usr/local/lib/python3.11/site-packages/apscheduler/schedulers/base.py", line 434, in add_job
    'trigger': self._create_trigger(trigger, trigger_args),
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/apscheduler/schedulers/base.py", line 935, in _create_trigger
    return self._create_plugin_instance('trigger', trigger, trigger_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/apscheduler/schedulers/base.py", line 920, in _create_plugin_instance
    return plugin_cls(**constructor_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: IntervalTrigger.__init__() got an unexpected keyword argument 'max_retries'
ERROR:cirrus.experiment_recipes:Failed to fetch recipes: 401 Client Error: Unauthorized for url: http://kinto:8888/v1/buckets/main/collections/nimbus-web-experiments/records
ERROR:cirrus.main:Failed to fetch recipes: 401 Client Error: Unauthorized for url: http://kinto:8888/v1/buckets/main/collections/nimbus-web-experiments/records
ERROR:apscheduler.executors.default:Job "fetch_schedule_recipes (trigger: interval[0:00:10], next run at: 2023-10-23 20:25:07 UTC)" raised an exception
Traceback (most recent call last):
  File "/app/cirrus/main.py", line 215, in fetch_schedule_recipes
    app.state.remote_setting.fetch_recipes()
  File "/app/cirrus/experiment_recipes.py", line 49, in fetch_recipes
    raise e
  File "/app/cirrus/experiment_recipes.py", line 46, in fetch_recipes
    response.raise_for_status()  # raises an HTTPError for bad status codes
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: http://kinto:8888/v1/buckets/main/collections/nimbus-web-experiments/records

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/apscheduler/executors/base_py3.py", line 30, in run_coroutine_job
    retval = await job.func(*job.args, **job.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/cirrus/main.py", line 219, in fetch_schedule_recipes
    app.state.scheduler.add_job(
  File "/usr/local/lib/python3.11/site-packages/apscheduler/schedulers/base.py", line 434, in add_job
    'trigger': self._create_trigger(trigger, trigger_args),
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/apscheduler/schedulers/base.py", line 935, in _create_trigger
    return self._create_plugin_instance('trigger', trigger, trigger_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/apscheduler/schedulers/base.py", line 920, in _create_plugin_instance
    return plugin_cls(**constructor_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: IntervalTrigger.__init__() got an unexpected keyword argument 'max_retries'
```

**Comment by yashikakhurana** [comment]
@jaredlockhart I think you did `make cirrus_up`, I would suggest do `make up` to stand rs

**Comment by yashikakhurana** [comment]
> @jaredlockhart I think you did `make cirrus_up`, I would suggest do `make up` to stand rs

or you can replace rs with stage link if you do not want to stand up whole thing, that's what I do generally

File: cirrus/server/Dockerfile [hunk 1]
```diff
@@ -70,10 +70,23 @@ COPY --from=rust-builder /application-services/fml.py /application-services/fml_
...
+USER app
 
-# Copy all other files into the container's working directory
-COPY . .
+# Copy all other files into the container's working directory (/app)
+COPY  . /app
 
 # Start application with Uvicorn server
-CMD ["uvicorn", "cirrus.main:app", "--host", "0.0.0.0", "--port", "8001"]
+CMD ["sh", "-c", "uvicorn cirrus.main:app --host 0.0.0.0 --port $PORT"]
```
Comment (jaredlockhart) [inline cirrus/server/Dockerfile:92]: Why are these changes necessary to add the healthchecks?

Comment (yashikakhurana) [reply cirrus/server/Dockerfile:92]: I have asked the same question to Brandon too, and waiting for his reply

Comment (jaredlockhart) [reply cirrus/server/Dockerfile:92]: I would expect the healthcheck pr to only add the endpoints and tests.  If these changes are necessary for some other reason I'd pull them into a separate PR.

Comment (yashikakhurana) [reply cirrus/server/Dockerfile]: I agree, so I took out the changes and this PR only contains changes related to end points, updated description as well.

**Review by jaredlockhart — APPROVED** [review]
Excellent 🎉 🎉 🎉 🎉
